src/gui/QvisExpressionsWindow.C

```diff
@@ -1240,62 +1247,72 @@ QvisExpressionsWindow::pyFilterSourceChanged()
     UpdatePythonExpression();
 }
 
-
 // ****************************************************************************
-//  Method:  QvisExpressionsWindow::nameTextChanged
+//  Method:  QvisExpressionsWindow::FinalizeExpressionNameChange
 //
 //  Purpose:
-//    Slot function when any change happens to the expression names.
-//
-//  Arguments:
-//    text       the new text.
-//
-//  Programmer:  Jeremy Meredith
-//  Creation:    October 10, 2004
-//
-//  Modifications:
-//    Jeremy Meredith, Mon Oct 25 12:40:31 PDT 2004
-//    Always access the expression list by index, just in case there
-//    are two expressions with the same name.
-//
-//    Jeremy Meredith, Mon Oct 17 10:42:08 PDT 2005
-//    Never allow an empty name to get into the expression list.  This
-//    could cause crashes.  ('6295)
-//
-//    Cyrus Harrison, Wed Jun 11 13:49:19 PDT 2008
-//    Initial Qt4 Port.
+//    When the user finishes editting the name of an expression in the gui,
+//    this function will check that name is unique. If it is not unique, then
+//    it will automatically update the name by appending a number to the end.
 //
+//  Programmer:  Eddie Rusu
+//  Creation:    Tue Jun 23 14:02:35 PDT 2020
 // ****************************************************************************
 void
-QvisExpressionsWindow::nameTextChanged(const QString &text)
+QvisExpressionsWindow::FinalizeExpressionNameChange()
 {
+    // Get the text from the currently selected item in the expression box
     int index = exprListBox->currentRow();
-
     if (index <  0)
         return;
+    QString text_from_expr_list = exprListBox->item(index)->text();
 
-    Expression &e = (*exprList)[indexMap[index]];
-
-    QString newname = text.trimmed();
-
-    if (newname.isEmpty())
+    // Get the text from the nameEdit field, automatically converting empty text to unnamed%1.
+    QString text_from_line_edit = nameEdit->text();
```

Comment:
Previously, text was trimmed, you didn't find it necessary? If spaces entered at the end of the name what happens?

Reply:
Awesome catch @biagas! I added the trimming.